src/Auth/OpenIDConnectService.php

```diff
@@ -0,0 +1,120 @@
+<?php
+namespace App\Auth;
+
+use Cake\Core\Configure;
+use Jumbojett\OpenIDConnectClient;
+
+class OpenIDConnectService
+{
+    private $oidc;
+
+    public function __construct()

```

Comment:
This should probably be a singleton. But that will mean we will have to be careful about switching access tokens for users. This will work for now.

Reply:
Can you provide a scenario where this would break in our current environment using our current procedures? How would we change those to 'be more careful'?

Reply:
It doesn't break anything but has different trade-offs:

**Per session instance (as implemented in this PR):**
This implementation means that there is a separate instance of `OpenIDConnectService` for every logged in user. Each of these instances will call the .well-known url (https://login.dallasmakerspace.org/auth/realms/DMS/.well-known/openid-configuration) on keycloak to get config. 

**Singleton approach:**
If we switch to singleton then there will be one instance for the whole application. We would avoid the call to get config for each user. However, we would need to maintain a map of access tokens for each user. Whenever we fetch groups for a user we have to set that user's access token on the oidc client.

 **Conclusion:**
 Singleton is too much of a hassle to implement without much benefit.